src/dmd/target.d

```diff
@@ -774,6 +776,90 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+
+    /**
+     * Get targetInfo by key
+     * Params:
+     *  name = name of targetInfo to get
+     *  loc = location to use for error messages
+     * Returns:
+     *  Expression for the requested targetInfo
+     */
+    extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
+    {
+        int ival;
+        const(char)[] sval;
+        switch (name.toDString)
+        {
+            case "arch":
+                sval = global.params.is64bit ? "x86_64" : "x86";
+                goto LstringExpr;
+            case "system":
+                if (global.params.isWindows)
+                    sval = "windows";
+                else if (global.params.isLinux)
+                    sval = "linux";
+                else if (global.params.isOSX)
+                    sval = "darwin";
+                else if (global.params.isFreeBSD)
+                    sval = "freebsd";
+                else if (global.params.isOpenBSD)
+                    sval = "openbsd";
+                else if (global.params.isDragonFlyBSD)
+                    sval = "dragonfly";
+                else if (global.params.isSolaris)
+                    sval = "solaris";
+                else
+                    sval = "unknown";
```

Comment:
Is it better to return `null`?

Reply:
perhaps this should just be an assert() since this case should be impossible unless DMD is broken somehow.

Reply:
Baremetal?

Reply:
I don't think DMD can do that. If all these are false, I expect DMD fails? (I doubt it's possible to invoke?)